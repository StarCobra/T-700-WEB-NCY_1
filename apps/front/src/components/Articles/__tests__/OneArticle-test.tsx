import React from "react";
import { act, create } from "react-test-renderer";
import Component from "../OneArticle";
import { MemoryRouter } from "react-router-dom";

describe("Components > Articles > OneArticle", () => {
  const tests = [
    {
      description: "generic",
      props: {},
    },
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;

      await act(async () => {
        root = create(
          <MemoryRouter>
            <Component {...test.props} />
          </MemoryRouter>,
        );
      });

      expect(root).toMatchSnapshot();
    });
  });
});
