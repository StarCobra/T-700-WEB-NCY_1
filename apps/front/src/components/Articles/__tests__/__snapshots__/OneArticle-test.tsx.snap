// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Articles > OneArticle should render the component generic 1`] = `
<div
  className="articleSample MuiBox-root css-0"
>
  <div
    className="articleThumbnail MuiBox-root css-0"
  >
    <img
      src="default"
    />
  </div>
  <div
    className="articleSampleContent MuiBox-root css-0"
  >
    <div
      className="articleTitle MuiBox-root css-0"
    >
      <h3>
        Bitcoin Collapse
      </h3>
    </div>
    <div
      className="articleSampleContent MuiBox-root css-0"
    >
      <p>
        In recent times, the once-revered and volatile king of cryptocurrencies, Bitcoin, has experienced a notable decline, leaving investors and enthusiasts puzzled. This downturn questions about the long-term sustainability of the digital currency and its ever-evolving landscape of finance and technology.
      </p>
    </div>
    <div
      className="articleDateSample MuiBox-root css-0"
    >
      <h4>
        November 13, 2023
      </h4>
    </div>
    <div
      className="seeMore MuiBox-root css-0"
    >
      <a
        className="Menu"
        href="/articles/1"
        id="menu1"
        onClick={[Function]}
      >
        Click here to see more ...
      </a>
    </div>
  </div>
</div>
`;
